GW2EIEvtcParser/EncounterLogic/Raids/W7/PeerlessQadim.cs

```diff
@@ -335,25 +386,195 @@ internal override void ComputePlayerCombatReplayActors(AbstractPlayer p, ParsedE
+                var hasPlayerDownedInAir = log.CombatData.GetDownEvents(p.AgentItem).Any(x => x.Time >= seg.Start && x.Time <= seg.End);
+                long magmaDropEnd = seg.End;
                 replay.AddDecorationWithGrowing(new CircleDecoration(magmaRadius, seg, "rgba(255, 50, 0, 0.2)", new AgentConnector(p)), magmaDropEnd);
-                Point3D position = p.GetCurrentInterpolatedPosition(log, magmaDropEnd);
-                if (position != null)
+                if (!log.CombatData.HasEffectData)
                 {
-                    string colorToUse = magmaColors[magmaColor];
-                    magmaColor = (magmaColor + 1) % 2;
-                    replay.AddDecorationWithGrowing(new CircleDecoration(magmaRadius, (magmaDropEnd, magmaDropEnd + magmaOffset), "rgba(" + colorToUse + ", 0.2)", new PositionConnector(position)), magmaDropEnd + magmaOffset);
-                    replay.Decorations.Add(new CircleDecoration(magmaRadius, (magmaDropEnd + magmaOffset, log.FightData.FightEnd), "rgba(" + colorToUse + ", 0.5)", new PositionConnector(position)));
+                    Point3D position = p.GetCurrentInterpolatedPosition(log, magmaDropEnd);
+                    if (position != null && !hasPlayerDownedInAir)
+                    {
+                        string colorToUse = magmaColors[magmaColor];
+                        magmaColor = (magmaColor + 1) % 2;
+                        long magmaWarningEnd = magmaDropEnd + magmaOffset;
+                        replay.AddDecorationWithGrowing(new CircleDecoration(magmaRadius, (magmaDropEnd, magmaWarningEnd), "rgba(" + colorToUse + ", 0.2)", new PositionConnector(position)), magmaDropEnd + magmaOffset);
+                        replay.Decorations.Add(new CircleDecoration(magmaRadius, (magmaWarningEnd, magmaWarningEnd + magmaDuration), "rgba(" + colorToUse + ", 0.5)", new PositionConnector(position)));
+                    }
                 }
             }
-            //sapping surge, bad red tether
+
+            // Sapping Surge, bad red tether
             AddTetherDecorations(log, p, replay, SappingSurge, "rgba(255, 0, 0, 0.4)");
-            // kinetic abundance, good (blue) tether
+            // Kinetic Abundance, good blue tether
             AddTetherDecorations(log, p, replay, KineticAbundance, "rgba(0, 255, 0, 0.4)");
+
+            // Add custom arrow overhead for the player lifted up
+            var castsLiftUp = p.GetCastEvents(log, log.FightData.FightStart, log.FightData.FightEnd).Where(x => x.SkillId == PlayerLiftUpQadimThePeerless).ToList();
+            var castsUnleash = p.GetCastEvents(log, log.FightData.FightStart, log.FightData.FightEnd).Where(x => x.SkillId == UnleashSAK).ToList();
+            var deadEvents = log.CombatData.GetDeadEvents(p.AgentItem).ToList();
+            foreach (AbstractCastEvent cast in castsLiftUp)
+            {
+                long timeUnleash = log.FightData.LogEnd;
```

Comment:
The FirstOrDefaults below can return null. You can replace it with:
`
long liftUpEnd = log.FightData.LogEnd;
AbstractCastEvent unleashCast = castsUnleash.Where(x => x.Time > cast.Time).FirstOrDefault();
if (unleashCast != null) {
   liftUpEnd = Math.Min(liftUpEnd , unleashCast.Time);
}
DeadEvent deadEvent = deadEvents.Where(x => x.Time > cast.Time).FirstOrDefault();
if (deadEvent != null) {
   liftUpEnd = Math.Min(liftUpEnd , deadEvent.Time);
}
`